Solutions/Marain.UserNotifications.Abstractions/Marain/Extensions/DependencyInjection/UserNotificationsSerializationServiceCollectionExtension.cs

```diff
@@ -0,0 +1,38 @@
+﻿// <copyright file="UserNotificationsSerializationServiceCollectionExtension.cs" company="Endjin Limited">
+// Copyright (c) Endjin Limited. All rights reserved.
+// </copyright>
+
+namespace Marain.Extensions.DependencyInjection
+{
+    using Corvus.ContentHandling;
+    using Marain.NotificationTemplates.CommunicationTemplates;
+    using Microsoft.Extensions.DependencyInjection;
+
+    /// <summary>
+    /// Adds content types to a service collection enabling serialization and deserializion of email, SMS, and web push templates.
+    /// </summary>
+    public static class UserNotificationsSerializationServiceCollectionExtension
+    {
+        /// <summary>
+        /// Adds user notification object types that include contentType
+        /// to the service collection.
+        /// </summary>
+        /// <param name="services">
+        /// The <see cref="IServiceCollection" /> to add the types to.
+        /// </param>
+        /// <returns>
+        /// The <see cref="IServiceCollection" />.
+        /// </returns>
+        public static IServiceCollection RegisterCoreUserNotificationsContentTypes(this IServiceCollection services)
+        {
+            services.AddContent(factory =>
+            {
+                factory.RegisterTransientContent<EmailTemplate>();
+                factory.RegisterTransientContent<SmsTemplate>();
+                factory.RegisterTransientContent<WebPushTemplate>();
+            });
```

Comment:
Change these registrations to use `RegisterContent` rather than `RegisterTransientContent`. This will allow you to use constructor-based deserialization for these types and avoid the issues commented on below to do with nullable properties.

Reply:
Changes have been made to use `RegisterContent`.